src/nvapi_disp.cpp

```diff
@@ -96,15 +89,85 @@ extern "C" {
+                    pHDRColorDataV1->mastering_display_data.displayWhitePoint_x = output->GetColorData().WhitePointX;
+                    pHDRColorDataV1->mastering_display_data.displayWhitePoint_y = output->GetColorData().WhitePointY;
+                    pHDRColorDataV1->mastering_display_data.max_display_mastering_luminance = output->GetColorData().MaxLuminance;
+                    pHDRColorDataV1->mastering_display_data.min_display_mastering_luminance = output->GetColorData().MinLuminance;
+                    pHDRColorDataV1->mastering_display_data.max_content_light_level = output->GetColorData().MaxLuminance;
+                    pHDRColorDataV1->mastering_display_data.max_frame_average_light_level = output->GetColorData().MaxFullFrameLuminance;
+                }
+            } else {
+                DXGI_COLOR_SPACE_TYPE colorspace = pHDRColorDataV1->hdrMode != NV_HDR_MODE_OFF ? DXGI_COLOR_SPACE_RGB_FULL_G2084_NONE_P2020 : DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709;
+                DXGI_HDR_METADATA_HDR10 metadata = {
+                    .RedPrimary = { pHDRColorDataV1->mastering_display_data.displayPrimary_x0, pHDRColorDataV1->mastering_display_data.displayPrimary_y0 },
+                    .GreenPrimary = { pHDRColorDataV1->mastering_display_data.displayPrimary_x1, pHDRColorDataV1->mastering_display_data.displayPrimary_y1 },
+                    .BluePrimary = { pHDRColorDataV1->mastering_display_data.displayPrimary_x2, pHDRColorDataV1->mastering_display_data.displayPrimary_y2 },
+                    .WhitePoint = { pHDRColorDataV1->mastering_display_data.displayWhitePoint_x, pHDRColorDataV1->mastering_display_data.displayWhitePoint_y },
+                    .MaxMasteringLuminance = pHDRColorDataV1->mastering_display_data.max_display_mastering_luminance,
+                    .MinMasteringLuminance = pHDRColorDataV1->mastering_display_data.min_display_mastering_luminance,
+                    .MaxContentLightLevel = pHDRColorDataV1->mastering_display_data.max_content_light_level,
+                    .MaxFrameAverageLightLevel = pHDRColorDataV1->mastering_display_data.max_frame_average_light_level,
+                };
+                if (FAILED(interop->SetGlobalHDRState(colorspace, &metadata)))
+                    return InvalidArgument(n);
+            }
+        } else {
+            return IncompatibleStructVersion(n);
         }
 
-        return NoImplementation(str::format(n, " (0x", std::hex, displayId, ")")); // See comment in NvAPI_Disp_GetHdrCapabilities
+        if (pHdrColorData->version == NV_HDR_COLOR_DATA_VER2) {
+            auto pHDRColorDataV2 = reinterpret_cast<NV_HDR_COLOR_DATA_V2*>(pHdrColorData);
+            if (pHDRColorDataV2->cmd == NV_HDR_CMD_GET) {
+                pHDRColorDataV2->hdrColorFormat = NV_COLOR_FORMAT_RGB;
+                pHDRColorDataV2->hdrDynamicRange = NV_DYNAMIC_RANGE_VESA; // No idea what this is meant to mean.
```

Comment:
- `NV_DYNAMIC_RANGE_VESA` is RGB full range
- `NV_DYNAMIC_RANGE_CEA` is RGB/YCbCr limited range
- `NV_DYNAMIC_RANGE_AUTO` automatically chooses something (probably reads some EDID/CTA-861 entries)

<https://github.com/Maassoft/ColorControl/blob/2c5b08af721362d28433e35397b0401a894a5d18/ColorControl/Services/NVIDIA/NvService.cs#L527>

> [Dynamic range: VESA (Full RGB) or CEA (Limited RGB/YUV)](https://github.com/Maassoft/ColorControl)

Reply:
VESA probably means that it is "IT" content which means it doesn't use timings from the CTA-861 block but timings from the normal EDID.
While CEA means that it is "CE" content which uses CTA definied timings.

EDID is more associated with RGB full because it is for PC monitors while CTA/CEA is for TVs which are more likely to expect limited range.

No idea what Nvidia was thinking with naming it like this...